lib/spack/spack/package.py

```diff
@@ -2839,23 +2878,24 @@ def test_process(pkg, kwargs):
 
                     # Run the tests
                     ran_actual_test_function = True
-                    test_fn(pkg)
+                    if verbose:
+                        with logger.force_echo():
+                            test_fn(pkg)
+                    else:
+                        test_fn(pkg)
```

Comment:
```suggestion
                    context = logger.force_echo if verbose else nullcontext
                    with context:
                        test_fn(pkg)
```

Do you think this would be easier or harder to maintain? Having two main entrypoints to the method gives me the willies, but I'm not sure this is better.